### ISSUE TITLE
Mack Trucks Campaign Header: Editable CTA with Form Fly-out functionality

### ISSUE THREAD

**Issue by KJohnsn** [hlxsites/vg-macktrucks-com#523]
**Description**
The[ Mack Trucks Campaign Microsites](https://info.macktrucks.com/incentives) have a form fly out which appears once a user clicks on the CTA. The form flies out from the right-hand side of the page and users can fill it in, submit it, and see a response in their email inbox.

The person who needs to have a new form added, will need to have it requested in Account Engagement, and then, once that's set up, the form can be created using Form Builder.

*Alternative to the flyout: embed the form in the page and the CTA contains the anchor link to take the user down to the form field within the page.

**Acceptance Criteria**

- [ ] The CTA activates the flyout field
- [ ] Inside the flyout, we can add references to other blocks (use case: form, text and button)
- [ ] The user is able to fill in the form easily
- [ ] The user is able to submit the form
- [ ] The CTA can be used in the header or inside content
- [ ] The CTA can be edited by the content author (label and styling)
- [ ] The header this CTA needs to be incorporated in is the one used on NHTSA
Document: https://adobe.sharepoint.com/:w:/r/sites/HelixProjects/_layouts/15/Doc.aspx?sourcedoc=%7B05DAC3A1-2EB9-445C-B487-CB9A180D3A5D%7D&file=black-header.docx&action=default&mobileredirect=true
Nav with CTA: https://adobe.sharepoint.com/:w:/r/sites/HelixProjects/_layouts/15/Doc.aspx?sourcedoc=%7BE407AA57-2DE0-48EC-9FD9-58CC3EB14CFC%7D&file=nav.docx&action=default&mobileredirect=true
- [ ] Fly out functionality can be disabled depending on the CTA being an anchor link in a page pointing to the relevant content area
- [ ] The user can easily create their form field spreadsheet and then once they click on the preview link, they can copy and paste that URL into the form excel sheet
- [ ] No special OneTrust requirements for this ticket - @ryanropero can you check this point, please?

**Screenshots**
![Image](https://github.com/hlxsites/vg-macktrucks-com/assets/126615590/8c9cc90c-d374-4f0b-a509-39775e43df45)

![Image](https://github.com/hlxsites/vg-macktrucks-com/assets/126615590/6f2332c9-8421-4bc0-bb98-1e08a200084e)

![Image](https://github.com/hlxsites/vg-macktrucks-com/assets/126615590/ca8ac6c1-9e01-45b5-8747-38156639aa2c)

Design Link: https://info.macktrucks.com/incentives

**Comment by Kannan-CM** [comment]
Hi Team,

Here are link to the pardot forms:

- **Make Paying Flexible – Flex Pay Promotion**

http://go.pardot.com/l/1038343/2024-01-19/3m6bz1

- **MD Series – MD Rate Selector**

http://go.pardot.com/l/1038343/2024-02-08/3m7jfl

- **Dependability Delivered – Mack Certified Used**

http://go.pardot.com/l/1038343/2024-02-08/3m7jhh

**Comment by BeckyMedlin** [comment]
@ryanropero There's a question regarding OneTrust in the description of this ticket. Please update the ticket with confirmation that nothing special is needed for OneTrust or additional info if something is needed.

Removed @WendyKruger because this ticket she has already given priority for it.

**Comment by ryanropero** [comment]
@BeckyMedlin I dont believe anything is needed for OneTrust for this form. They only track submissions. Views are anonymous and there isn't any personal data

**Comment by BeckyMedlin** [comment]
@mollyaljenkins Please re-refine and see if this ticket can be broken down further (split into at least 2 tickets).

**Comment by BeckyMedlin** [comment]
@WendyKruger, @jrcooke12, and @ZoeBeykirch Is this ticket still valid?
